### PR TITLE
BOLT 07: copy edit, part 2

### PR DESCRIPTION
**Pull request by toadlyBroodle**
Complete first pass copy edit of second half of BOLT 07.
 - imposed 'origin/final node' terminology
 - one FIXME question

### CODE REVIEW

File: 07-routing-gossip.md [hunk 1]
```diff
@@ -308,33 +308,31 @@ The receiving node:
...
+optional) ones will have _odd_ `feature` _bits_, incompatible ones will have
+_even_ `feature` _bits_. These may be propagated by nodes even if they
+cannot process the announcements themselves.
 
-New address types can be added in the future; as address descriptors have
+New address types may be added in the future; as address descriptors have
 to be ordered in ascending order, unknown ones can be safely ignored.
-Future fields beyond `addresses` can still be added, optionally with
-padding within `addresses` if they require certain alignment.
+Additional fields beyond `addresses` may also be added in the future—with
```
Comment (practicalswift) [inline 07-routing-gossip.md]: Nit: "future – with" instead of "future–with"?

Comment (toadlyBroodle) [reply 07-routing-gossip.md]: I think this is a matter of preference, so far I've been using without spaces, can change it, but most important thing I think is to maintain consistency.

Comment (practicalswift) [reply 07-routing-gossip.md]: Agreed. As long as it is consistent I'm fine :-)

File: 07-routing-gossip.md [hunk 2]
```diff
@@ -226,10 +226,9 @@ nodes not associated with an already known channel are ignored.
    * [`2`:`addrlen`]
    * [`addrlen`:`addresses`]
 
-The `timestamp` allows ordering of messages, in the case of multiple
-announcements; the `rgb_color` and `alias` allow intelligence services to give
-their nodes cool monikers like 'IRATEMONK' and 'WISTFULTOLL' and to use the
-color black.
+`timestamp` allows for the ordering of messages, in the case of multiple
+announcements. `rgb_color` and `alias` allow intelligence services to assign
+nodes colors (e.g. black) and cool monikers like 'IRATEMONK' and 'WISTFULTOLL'.
```
Comment (shannona) [inline 07-routing-gossip.md]: It's a niggling bit, but I always prefer parallel lists (e.g., "assign nodes colors like black and ..."

File: 07-routing-gossip.md [hunk 3]
```diff
@@ -379,144 +371,231 @@ Note that such a `channel_update` that is not preceded by a `channel_announcemen
...
+  - MUST set `signature` to the signature of the double-SHA256 of the entire
+  remaining packet after `signature`, using its own `node_id`.
+  - MUST set `chain_hash` AND `short_channel_id` to match the 32-byte hash AND
+  8-byte channel ID that uniquely identifies the channel specified in the
+  `channel_announcement` message.
+  - if the origin node is `node_id_1` in the message:
+    - MUST set the `direction` bit of `flags` to 0.
+  - otherwise:
+    - MUST set the `direction` bit of `flags` to 1.
+  - MUST set bits which are not assigned a meaning to 0.
```
Comment (shannona) [inline 07-routing-gossip.md]: that are not assigned a meaning

File: 07-routing-gossip.md [hunk 4]
```diff
@@ -379,144 +371,231 @@ Note that such a `channel_update` that is not preceded by a `channel_announcemen
...
+    - Note: this is discouraged, in order to keep the resource requirements low.
+  - upon connection establishment:
+    - SHOULD send all `channel_announcement` messages, followed by the latest
+    `node_announcement` AND `channel_update` messages.
 
 ### Rationale
 
-Batching announcements forms a natural rate limit with low overhead.
+Once the announcement has been processed, it's added to a list of outgoing
+announcements, destined for the processing node's peers, and replaces any older
```
Comment (shannona) [inline 07-routing-gossip.md]: Would "replacing any" be clearer? Right now you could read that as a separate action and I think it's the same.

**Comment by practicalswift** [comment]
ACK 9cd03366fe8f7e1e9b6099ef57e5c2cc6bb608f9

File: 07-routing-gossip.md (hunk 2 above)
Comment (toadlyBroodle) [reply 07-routing-gossip.md]: your wish is my command

File: 07-routing-gossip.md (hunk 3 above)
Comment (toadlyBroodle) [reply 07-routing-gossip.md]: fixed

File: 07-routing-gossip.md (hunk 4 above)
Comment (toadlyBroodle) [reply 07-routing-gossip.md]: done

**Pull request merged**